nix/README.md

```diff
@@ -66,17 +68,31 @@ The PostgREST utilities available in `nix-shell` all have names that begin with
 ```
 # Note: The utilities listed here might not be up to date.
 [nix-shell]$ postgrest-<tab>
-postgrest-lint                         postgrest-test-spec-postgresql-10
-postgrest-style                        postgrest-test-spec-postgresql-11
-postgrest-style-check                  postgrest-test-spec-postgresql-12
-postgrest-test-all                     postgrest-test-spec-postgresql-9.4
-postgrest-test-spec                    postgrest-test-spec-postgresql-9.5
-postgrest-test-spec-all                postgrest-test-spec-postgresql-9.6
+postgrest-lint         postgrest-style        postgrest-style-check
 
 [nix-shell]$
 
 ```
 
+Some additional modules like `tests`, `docker` and `release` have large
+dependencies that would need to be built before the shell becomes available,
```

Comment:
Hmm.. couldn't we have the end-to-end tests(running io-tests.sh and memory-tests.sh) in another nix file(like `io-tests.nix`) and avoid building the static and profiled builds on nix-shell entrance?

I was really liking the DX of going inside nix-shell and being able to do `postgrest-test-<tab>`.

Reply:
Then we could do `nix-shell --arg io-tests true` in case we need to run the io tests. I think this would be better because those tests aren't run that frequently.

Reply:
You're right, that would be much nicer! Didn't think about that.

With the new commit the spec tests are available in the nix-shell by default and the other tests can be enabled with flags!

Reply:
@monacoremo Thanks! Much better now.

Just something I noticed. Looks like the nix memory tests consume less memory than the stack-test-memory.

nix-tests:

```
Running memory usage tests..
ok 1 - POST /rpc/leak?columns=blob: with a json key of 1M the memory usage(9,075,584 bytes) is less than 13M
ok 2 - POST /leak?columns=blob: with a json key of 1M the memory usage(9,193,768 bytes) is less than 13M
```
stack-memory-test:
```
ok 1 - POST /rpc/leak?columns=blob: with a json key of 1M the memory usage(12,122,480 bytes) is less than 13M
ok 2 - POST /leak?columns=blob: with a json key of 1M the memory usage(12,257,056 bytes) is less than 13M
```

Not sure why. But it's good nonetheless, the stack memory tests used to break from time to time.

Reply:
@steve-chavez Interesting... I don't have much experience with profiling haskell, might it be that some libraries are not included in the nix profiled build? Or different optimization levels?